multimap/src/main/java/org/infinispan/multimap/impl/function/set/SPopFunction.java

```diff
@@ -0,0 +1,92 @@
+package org.infinispan.multimap.impl.function.set;
+
+import java.io.IOException;
+import java.io.ObjectInput;
+import java.io.ObjectOutput;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.Random;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.infinispan.commons.marshall.AdvancedExternalizer;
+import org.infinispan.functional.EntryView;
+import org.infinispan.multimap.impl.ExternalizerIds;
+import org.infinispan.multimap.impl.SetBucket;
+
+/**
+ * Serializable function used by
+ * {@link org.infinispan.multimap.impl.EmbeddedSetCache#remove}
+ * to remove elements to a Set.
+ *
+ * @author Vittorio Rigamonti
+ * @see <a href="http://infinispan.org/documentation/">Marshalling of
+ *      Functions</a>
+ * @since 15.0
+ */
+public final class SPopFunction<K, V> implements SetBucketBaseFunction<K, V, Collection<V>> {
+   public static final AdvancedExternalizer<SPopFunction> EXTERNALIZER = new Externalizer();
+   private final Long count;
+   private final boolean remove;
+
+   public SPopFunction(Long count, boolean remove) {
+      this.count = count;
+      this.remove = remove;
+   }
+
+   @Override
+   public Collection<V> apply(EntryView.ReadWriteEntryView<K, SetBucket<V>> entryView) {
+      Optional<SetBucket<V>> existing = entryView.peek();
+      if (!existing.isPresent() || count == 0) {
```

Comment:
Since we treat non-existent entries like this, we could return an empty list when `count == 0` without invoking the function.

Reply:
optimized the condition